lib/internal/test_runner/utils.js

```diff
@@ -164,10 +166,14 @@ async function setupTestReporters(testsStream) {
                                     'must match the number of specified \'--test-reporter-destination\'');
   }
 
-  const reportersMap = await getReportersMap(reporters, destinations);
-  for (let i = 0; i < reportersMap.length; i++) {
-    const { reporter, destination } = reportersMap[i];
-    compose(testsStream, reporter).pipe(destination);
+  try {
+    const reportersMap = await getReportersMap(reporters, destinations);
+    for (let i = 0; i < reportersMap.length; i++) {
+      const { reporter, destination } = reportersMap[i];
+      compose(testsStream, reporter).pipe(destination);
+    }
+  } catch (err) {
+    throw new ERR_TEST_FAILURE(err, kAsyncBootstrapFailure);
```

Comment:
nit: we should probably only catch async errors, and not errors that might happen in the for loop.
```suggestion
  const reportersMap = await PromisePrototypeThen(
    getReportersMap(reporters, destinations), undefined,
    (err) => PromiseReject(new ERR_TEST_FAILURE(err, kAsyncBootstrapFailure)),
  );
  for (let i = 0; i < reportersMap.length; i++) {
    const { reporter, destination } = reportersMap[i];
    compose(testsStream, reporter).pipe(destination);
```

Reply:
By the time this code is reached, `getReportersMap()` has made everything async from the test runner's point of view.